dev/io.openliberty.session.6.1.internal/src/io/openliberty/session/AbstractSessionData61.java

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.session;
+
+import java.util.function.Consumer;
+import java.util.logging.Level;
+
+import com.ibm.ejs.ras.TraceNLS;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.session.AbstractSessionData;
+import com.ibm.ws.session.SessionContext;
+import com.ibm.ws.session.utils.LoggingUtil;
+import com.ibm.wsspi.session.ISession;
+
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.http.HttpSession;
+
+/*
+ * Servlet 6.1 - support HttpSession getAccessor() API
+ * 
+ * @see jakarta.servlet.http.HttpSession#getAccessor()
+ * @see jakarta.servlet.http.HttpSession.Accessor#access(Consumer<HttpSession>)
+ */
+public abstract class AbstractSessionData61 extends AbstractSessionData {
+    private static final String CLASS_NAME = "AbstractSessionData61";
+    private static TraceNLS nls = TraceNLS.getTraceNLS(AbstractSessionData61.class, "io.openliberty.session.resources.SessionMessages");
+
+    protected AbstractSessionData61(ISession session) {
+        super(session);
+
+        if (TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINE)) {
+            LoggingUtil.SESSION_LOGGER_CORE.log(Level.FINE, CLASS_NAME + " Constructor (ISession) [" + session + "]");
+        }
+    }
+
+    protected AbstractSessionData61(ISession session, SessionContext sessCtx, ServletContext servCtx) {
+        super(session, sessCtx, servCtx);
+
+        if (TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINER)) {
+            LoggingUtil.SESSION_LOGGER_CORE.log(Level.FINE, CLASS_NAME + " Constructor , ISession [ " + session + " ] ; ServletContext [" + servCtx + "]");
+        }
+    }
+
+    public class SessionDataAccessor implements HttpSession.Accessor {
```

Comment:
```suggestion
    private static class SessionDataAccessor implements HttpSession.Accessor {
```
From what I can see it doesn't need to be public or non-static.